src/history/database.py

```
@@ -131,11 +184,125 @@ def initialize(self) -> None:
+        #    the damage is severe enough that integrity_check can't
+        #    even walk the pages. THESE are corruption signals —
+        #    normalize to our typed exception.
+        try:
+            cursor = conn.execute("PRAGMA integrity_check")
+            rows = cursor.fetchall()
+        except sqlite3.OperationalError:
+            # Transient (lock/busy/I-O). Propagate unchanged — the
+            # file is most likely healthy; the caller can retry.
+            raise
+        except sqlite3.DatabaseError as exc:
+            logger.error(
```

Ruff (0.15.11)

[warning] Logging `.exception(...)` should be used instead of `.error(..., exc_info=True)`

(G201)